Sources/Fullscreen/SearchResultMapView/SearchResultMapView.swift

```diff
@@ -0,0 +1,186 @@
+        mapView.setRegion(region, animated: animated)
+    }
+
+    public func setCenterOnUserLocation(regionDistance: CLLocationDegrees, animated: Bool) {
+        if mapView.userLocation.location != nil {
+            setCenter(mapView.userLocation.coordinate, regionDistance: regionDistance, animated: animated)
+        }
+    }
+
+    public func setMapOverlay(_ newOverlay: MKTileOverlay) {
+        if let lastOverlay = mapView.overlays.last {
+            mapView.removeOverlay(lastOverlay)
+        }
+
+        mapView.addOverlay(newOverlay)
+    }
+
+    public func clearMapOverlay(_ overlay: MKTileOverlay) {
+        mapView.removeOverlay(overlay)
+    }
+
+    public func addAnnotation(_ annotation: SearchResultMapViewAnnotation) {
+        mapView.addAnnotation(annotation)
+    }
+
+    public func clearAnnotations() {
+        mapView.removeAnnotations(mapView.annotations)
+    }
+
+    public func focusAnnotations() {
+        let annotationsToShow = mapView.annotations.filter { $0 is SearchResultMapViewAnnotation }
+        mapView.showAnnotations(annotationsToShow, animated: true)
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+        backgroundColor = .milk
+
+        addSubview(mapView)
+
+        mapView.fillInSuperview()
+        mapView.addSubview(mapSettingsButton)
+
+        NSLayoutConstraint.activate([
+            mapSettingsButton.topAnchor.constraint(equalTo: mapView.topAnchor, constant: 8),
+            mapSettingsButton.trailingAnchor.constraint(equalTo: mapView.trailingAnchor, constant: -8)
+            ])
+    }
+
+}
+
+// MARK: - Extensions
+
+extension SearchResultMapView: MapSettingsButtonDelegate {
+
+    public func mapSettingsButtonDidSelectChangeMapTypeButton(_ view: MapSettingsButton) {
+        delegate?.searchResultMapViewDidSelectChangeMapTypeButton(self, mapSettingsButtonView: view)
+    }
+
+    public func mapSettingsButtonDidSelectCenterMapButton(_ view: MapSettingsButton) {
+        delegate?.searchResultMapViewDidSelectCenterMapButton(self)
+    }
+
+}
+
+extension SearchResultMapView: MKMapViewDelegate {
+
+    public func mapView(_ mapView: MKMapView, rendererFor overlay: MKOverlay) -> MKOverlayRenderer {
+        if let overlay = overlay as? MKTileOverlay {
+            return MKTileOverlayRenderer(tileOverlay: overlay)
+        }
+
+        return MKOverlayRenderer(overlay: overlay)
+    }
+
+    public func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+        guard !(annotation is MKUserLocation) else { return nil }
+
+        if let annotation = annotation as? SearchResultMapViewAnnotation {
+            let marker = MKAnnotationView(annotation: annotation, reuseIdentifier: annotation.isCluster ? "clusterPOI" : "POI")
+            marker.image = annotation.image
+            marker.accessibilityIdentifier = "annotationPOI"
```

Comment:
What's this needed for? Is this what Siri will read in VoiceOver?

Reply:
It can probably be removed, since I most likely just added this line since it was present in the old code. As far as I can understand, this is used to identify an annotation element in the  existing UITests.